partiql-eval/src/main/kotlin/org/partiql/eval/internal/operator/rel/RelExclude.kt

```diff
@@ -0,0 +1,212 @@
+package org.partiql.eval.internal.operator.rel
+
+import org.partiql.eval.internal.Record
+import org.partiql.eval.internal.exclude.CompiledExcludeItem
+import org.partiql.eval.internal.exclude.ExcludeFieldCase
+import org.partiql.eval.internal.exclude.ExcludeNode
+import org.partiql.eval.internal.exclude.ExcludeStep
+import org.partiql.eval.internal.operator.Operator
+import org.partiql.plan.Identifier
+import org.partiql.plan.Rel
+import org.partiql.value.BagValue
+import org.partiql.value.CollectionValue
+import org.partiql.value.ListValue
+import org.partiql.value.PartiQLValue
+import org.partiql.value.PartiQLValueExperimental
+import org.partiql.value.PartiQLValueType
+import org.partiql.value.SexpValue
+import org.partiql.value.StructValue
+import org.partiql.value.bagValue
+import org.partiql.value.listValue
+import org.partiql.value.sexpValue
+import org.partiql.value.structValue
+
+internal class RelExclude(
+    val input: Operator.Relation,
+    private val compiledExcludeItems: List<CompiledExcludeItem>
+) : Operator.Relation {
+
+    override fun open() {
+        input.open()
+    }
+
+    override fun next(): Record? {
+        while (true) {
+            val row = input.next() ?: return null
+            val newRecord = compiledExcludeItems.fold(row) { curRecord, expr ->
+                excludeOnRecord(curRecord, expr)
+            }
+            return newRecord
+        }
+    }
+
+    override fun close() {
+        input.close()
+    }
+}
+
+@OptIn(PartiQLValueExperimental::class)
```

Comment:
Rather than having a bunch of top-level functions, could you please place these in the appropriate spots? Compile related functions in the Compiler, and evaluation related items wihtin the RelExclude operator?

Reply:
Changed locations of the top-level functions in https://github.com/partiql/partiql-lang-kotlin/pull/1320/commits/9e9a04032dc8c57a17bbbcb624bba166fd0ecdd0